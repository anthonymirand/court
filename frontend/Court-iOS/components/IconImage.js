import React from 'react';

import Mask from 'react-native-mask';
import { Image, StyleSheet, View } from 'react-native';
import Icons from '../constants/Icons';

export default class IconImage extends React.Component {
  render() {
    const { size, avatar, color } = this.props;
    return (
      <View style={{height: size, width: size, backgroundColor: color, justifyContent: 'center', alignItems: 'center'}}>
        <Image
          style={{ width: .6 * size, height: .6 * size, tintColor: 'white' }}
          source={ Icons[avatar] }
        />
      </View>
    );
  }
}
