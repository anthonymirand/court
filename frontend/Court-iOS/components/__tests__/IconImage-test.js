import 'react-native';
import React from 'react';
import IconImage from '../IconImage';
import renderer from 'react-test-renderer';

it('Renders Default IconImage', () => {
  const tree = renderer.create(<IconImage size={100} avatar="sloth" color="#000"/>).toJSON();

  expect(tree).toMatchSnapshot();
});
