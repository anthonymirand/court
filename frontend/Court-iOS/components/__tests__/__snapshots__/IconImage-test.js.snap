// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default IconImage 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      },
      Object {},
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 60,
        "tintColor": "white",
        "width": 60,
      }
    }
  />
</View>
`;
